packages/buckets/src/buckets.ts

```diff
@@ -54,6 +70,8 @@ const logger = log.getLogger('buckets')
  * ```typescript
  * import { Buckets } from '@textile/hub'
  *
+ * // This method requires that you already specify the Thread containing
+ * // the bucket with buckets.withThread(<thread name>).
```

Comment:
Wondering if we should model this more like the Go lib, and actually include the with* in the constructor?

```
const buckets = new Buckets(withThread(...))
```

Maybe more difficult here because we might not already know the Thread, but this would be optional anyway.

Reply:
We never call 'new Buckets(...)` for anything though. you almost always call, `Buckets.withUserAuth(...)` which means you must set the threadId after in almost all cases. and even more obvious is to just call the `getOrInit` which never makes you deal with the thread. i just wanted to include the note where you are commenting in case people go down this adventurous path